astartes/samplers/interpolation/kennardstone.py

```diff
@@ -18,17 +19,9 @@ def _ks_split(self):
         set, so we ask for all but one of the points, and then put that
         index into the list at the end to circumvent this.
         """
-        _, _, samples_idxs, spare_idx = train_test_split(
+        _, _, samples_idxs, spare_idx = ks_train_test_split(
             self.X,
```

Comment:
I believe we can remove self.X and only pass the list of indices (i.e. `list(range(len(self.X)))`) since splitting the X object is simply assigned to the throw away variables of `_`.

Reply:
The Kennard-Stone algorithm chooses the order of the samples based on the X array (sampling the extremes into the train set, and then interpolating for the test set) so self.X is required -- we ignore it because we ultimately just want to set _samples_idxs to indicate the order by which the samples should be withdrawn.

Throwing out the sampled X array is wasted work since we end up splitting it again, but the cost of doing so is low compared to the runtime of actually deciding on the sampling order.

Reply:
My mistake. That makes sense. Thanks for explaining!